### PR TITLE
Feature/hilbert station init 51

### LINKED ISSUE #51: Do not redeploy on stations after configuration synchronization if nothing changed
If the hilbert configuration is modified, it will be synchronized will all stations and the system will be restarted on all stations. However, the configuration change might not effect each station and if the configuration of a station didn't change, the station should continue running without interruption.

### PR DESCRIPTION
**Pull request by malex984**
This PR contains major work on `hilbert-station` bash script: many internal fixes and improvements.

In particular it ensures that configuration installation/initialization/preparation command (`init`) will detect changes between the current and the new configuration and react appropriately, while trying to keep the currently running top application in place. Thus it also fixes hilbert/hilbert-cli#51

### CODE REVIEW

File: tools/hilbert-station [hunk 1]
```diff
@@ -3,24 +3,89 @@
...
+# According to the above table, exit codes 1 - 2, 126 - 165, and 255 have special meanings,
+# and should therefore be avoided for user-specified exit parameters.
+# Ending a script with exit 127 would certainly cause confusion when troubleshooting
+# (is the error code a "command not found" or a user-defined one?).
+# However, many scripts use an exit 1 as a general bailout-upon-error.
+# Since exit code 1 signifies so many possible errors, it is not particularly useful in debugging.
+
+
+
+export RSYNC=${RSYNC:-rsync}
```
Comment (malex984) [inline tools/hilbert-station]: maybe use `declare -rg RSYNC=${HILBERT_DEPENDENCY_RSYNC:-rsync}`

File: tools/hilbert-station [hunk 2]
```diff
@@ -64,66 +129,115 @@ trap 'exec 2>&4 1>&3 ' SIGHUP SIGINT SIGQUIT SIGPIPE SIGTERM EXIT RETURN
...
         ERROR "Another user is already running ${TOOL}: please try again later!"
-        exit 2
+        exit ${ERR_CODE_EXT_ERROR}
+    fi
+    return $?
+}
+
+## The following is due to https://en.wikipedia.org/wiki/ANSI_escape_code
+#declare -rg ANSI_GREEN='\033[0;32m'
+declare -rg ANSI_GRAY='\033[0;37m'
```
Comment (malex984) [inline tools/hilbert-station]: NOTE: try to detect support for colors in current terminal

File: tools/hilbert-station [hunk 3]
```diff
@@ -64,66 +129,115 @@ trap 'exec 2>&4 1>&3 ' SIGHUP SIGINT SIGQUIT SIGPIPE SIGTERM EXIT RETURN
...
+declare -rg ANSI_CYAN='\033[1;36m'
+declare -rg ANSI_YELLOW='\033[1;33m'
+declare -rg ANSI_RED='\033[1;31m'
+declare -rg ANSI_BRIGHT_MAGENTA='\033[1;95m'
+declare -rg ANSI_BRIGHT_WHITE='\033[1;97m'
+declare -rg ANSI_NO_COLOR='\033[0m'
+
+# default colors for the logging routines:
+declare -rg COLOR_DEBUG="${COLOR_DEBUG:-${ANSI_GRAY}}"
```
Comment (malex984) [inline tools/hilbert-station]: maybe use `HILBERT_CONSOLE_COLOR_DEBUG` as above?

File: tools/hilbert-station (hunk 2 above)
Comment (porst17) [reply tools/hilbert-station]: https://unix.stackexchange.com/questions/9957/how-to-check-if-bash-can-print-colors

Comment (porst17) [reply tools/hilbert-station]: `tput colors`

**Review by porst17 — APPROVED** [review]

**Review by porst17 — CHANGES_REQUESTED** [review]
I overlooked that there is still changes missing:
- there needs to be a safeguard for the ANSI colors on the terminal. Ask `tput colors` about it and if no colors are supported, unset/empty the `ANSI_*` color variables
- we wanted to name the dependencies with `HILBERT_DEPENDENCY_*`
- you wanted to do something with `HILBERT_CONSOLE_COLOR_DEBUG`

**Comment by malex984** [comment]
@porst17 please check new commits ece5022 and 2345b5a for fixes due to 
https://github.com/hilbert/hilbert-cli/pull/73#pullrequestreview-96097318

**Review by porst17 — APPROVED** [review]
Looks good to me now.

**Comment by porst17** [comment]
Anything missing here? Otherwise, please merge.

**Review by porst17 — APPROVED** [review]

**Comment by malex984** [comment]
@porst17 i fixed one major bug and a few minor ones. the current version works well on bigfoot60. i pushed all the changes here since they are all about the same script and related functionality.

**Review by porst17 — APPROVED** [review]

**Pull request merged**